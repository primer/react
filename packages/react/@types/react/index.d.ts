// eslint-disable-next-line no-restricted-imports
export * from 'react'

declare module 'react' {
  interface CSSProperties {
    [key: `--${string}`]: string | number | undefined
  }
}
