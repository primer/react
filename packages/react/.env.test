VITEST_SEGFAULT_RETRY=0
VITEST_SKIP_BROWSER=1