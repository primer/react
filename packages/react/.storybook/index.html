<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Primer Preview Page</title>
  <link href="https://unpkg.com/@primer/css@^20.2.4/dist/primer.css" rel="stylesheet" />
</head>
<body class="p-3">
  <div class="Header">
    <h1>Primer Preview Page</h1>
  </div>
  <div class="p-2 m-2 d-flex flex-column">
    <ul class="filter-list">
      <li><a class="filter-item" href="./storybook" class="btn-link mb-2">Storybook</a></li>
      <li><a class="filter-item" href="./vrt-no-flag" class="btn-link mb-2">Visual Regression Testing</a></li>
      <li><a class="filter-item" href="./vrt-all-flags" class="btn-link mb-2">Visual Regression Testing (All Flags
          Enabled)</a></li>
      <li><a class="filter-item" href="./aat-no-flag" class="btn-link mb-2">Automated Accessibility Testing</a></li>
      <li><a class="filter-item" href="./aat-all-flags" class="btn-link mb-2">Automated Accessibility Testing (All Flags
          Enabled)</a></li>
    </ul>
  </div>
</body>
</html>