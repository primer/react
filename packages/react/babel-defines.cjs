const shared = {
  __DEV__: "process.env.NODE_ENV !== 'production'",
}

module.exports = {
  development: shared,
  test: shared,
  production: {
    ...shared,
    __DEV__: 'false',
    'process.env.NODE_ENV': "'production'",
  },
}
