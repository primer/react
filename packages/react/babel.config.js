const defines = require('./babel-defines')
const {isSupported} = require('./script/react-compiler.mjs')

function replacementPlugin(env) {
  return ['babel-plugin-transform-replace-expressions', {replace: defines[env]}]
}

const sharedPlugins = [
  [
    'babel-plugin-react-compiler',
    {
      target: '18',
      sources: isSupported,
    },
  ],
  'macros',
  'dev-expression',
  'add-react-displayname',
  'babel-plugin-styled-components',
  '@babel/plugin-proposal-nullish-coalescing-operator',
  '@babel/plugin-proposal-optional-chaining',
]

function makePresets(moduleValue) {
  return [
    '@babel/preset-typescript',
    [
      '@babel/preset-react',
      {
        modules: moduleValue,
        runtime: 'automatic',
      },
    ],
  ]
}

module.exports = {
  env: {
    development: {
      presets: makePresets(process.env.BABEL_MODULE || false),
      plugins: [...sharedPlugins, replacementPlugin('development')],
    },
    production: {
      presets: makePresets(false),
      plugins: [...sharedPlugins, replacementPlugin('production')],
    },
    test: {
      presets: makePresets('commonjs'),
      plugins: [...sharedPlugins, ['@babel/plugin-transform-modules-commonjs'], replacementPlugin('test')],
    },
  },
}
