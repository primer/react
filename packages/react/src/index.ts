'use client'

export {default as theme} from './theme'
export {get as themeGet} from './constants'
export {default as BaseStyles} from './BaseStyles'
export type {BaseStylesProps} from './BaseStyles'
export {default as ThemeProvider, useTheme, useColorSchemeVar} from './ThemeProvider'
export type {ThemeProviderProps} from './ThemeProvider'

// Layout
export {default as Box} from './Box'
export type {BoxProps} from './Box'
export * from './Button'
export {PageLayout} from './PageLayout'
export type {
  PageLayoutProps,
  PageLayoutHeaderProps,
  PageLayoutContentProps,
  PageLayoutPaneProps,
  PageLayoutFooterProps,
} from './PageLayout'
export {SplitPageLayout} from './SplitPageLayout'
export type {
  SplitPageLayoutProps,
  SplitPageLayoutHeaderProps,
  SplitPageLayoutContentProps,
  SplitPageLayoutPaneProps,
  SplitPageLayoutFooterProps,
} from './SplitPageLayout'

// Hooks
export {default as useDetails} from './hooks/useDetails'
export {default as useSafeTimeout} from './hooks/useSafeTimeout'
export {useOnOutsideClick} from './hooks/useOnOutsideClick'
export type {TouchOrMouseEvent} from './hooks/useOnOutsideClick'
export {useOpenAndCloseFocus} from './hooks/useOpenAndCloseFocus'
export {useOnEscapePress} from './hooks/useOnEscapePress'
export {useOverlay} from './hooks/useOverlay'
export {useConfirm} from './ConfirmationDialog/ConfirmationDialog'
export {useFocusTrap} from './hooks/useFocusTrap'
export type {FocusTrapHookSettings} from './hooks/useFocusTrap'
export {useFocusZone} from './hooks/useFocusZone'
export type {FocusZoneHookSettings} from './hooks/useFocusZone'
export {useRefObjectAsForwardedRef} from './hooks/useRefObjectAsForwardedRef'
export {useResizeObserver} from './hooks/useResizeObserver'
export {useResponsiveValue} from './hooks/useResponsiveValue'

// Utils
export {createComponent} from './utils/create-component'

// Components
export {default as Radio} from './Radio'
export type {RadioProps} from './Radio'
export {ActionList} from './ActionList'
export type {
  ActionListProps,
  ActionListGroupProps,
  ActionListItemProps,
  ActionListLinkItemProps,
  ActionListDividerProps,
  ActionListDescriptionProps,
  ActionListLeadingVisualProps,
  ActionListTrailingVisualProps,
} from './ActionList'
export {ActionMenu} from './ActionMenu'
export type {ActionMenuProps, ActionMenuAnchorProps, ActionMenuButtonProps} from './ActionMenu'
export {AnchoredOverlay} from './AnchoredOverlay'
export type {AnchoredOverlayProps} from './AnchoredOverlay'
export {default as Autocomplete} from './Autocomplete'
export type {AutocompleteMenuProps, AutocompleteInputProps, AutocompleteOverlayProps} from './Autocomplete'
export {default as Avatar} from './Avatar'
export type {AvatarProps} from './Avatar'
export {default as AvatarPair} from './AvatarPair'
export type {AvatarPairProps} from './AvatarPair'
export {default as AvatarStack} from './AvatarStack'
export type {AvatarStackProps} from './AvatarStack'
export {default as BranchName} from './BranchName'
export type {BranchNameProps} from './BranchName'
export {default as Breadcrumbs, Breadcrumb} from './Breadcrumbs'
export type {BreadcrumbsProps, BreadcrumbsItemProps, BreadcrumbProps, BreadcrumbItemProps} from './Breadcrumbs'
export {default as ButtonGroup} from './ButtonGroup'
export type {ButtonGroupProps} from './ButtonGroup'
export {default as Caret} from './Caret'
export type {CircleBadgeProps, CircleBadgeIconProps} from './CircleBadge'
export {default as CircleOcticon} from './CircleOcticon'
export type {CircleOcticonProps} from './CircleOcticon'
export {default as CheckboxGroup} from './CheckboxGroup'
export type {CaretProps} from './Caret'
export {default as CircleBadge} from './CircleBadge'
export {default as CounterLabel} from './CounterLabel'
export type {CounterLabelProps} from './CounterLabel'
export {default as Details} from './Details'
export type {DetailsProps} from './Details'
export {default as Dialog} from './Dialog'
export type {DialogProps, DialogHeaderProps} from './Dialog'
export type {ConfirmationDialogProps} from './ConfirmationDialog/ConfirmationDialog'
export {ConfirmationDialog} from './ConfirmationDialog/ConfirmationDialog'
export {default as Flash} from './Flash'
export type {FlashProps} from './Flash'
export {default as FormControl} from './FormControl'
export {useFormControlForwardedProps} from './FormControl'
export {default as Header} from './Header'
export type {HeaderProps, HeaderItemProps, HeaderLinkProps} from './Header'
export {default as Heading} from './Heading'
export type {HeadingProps} from './Heading'
export {default as Label} from './Label'
export type {LabelProps} from './Label'
export {default as LabelGroup} from './LabelGroup'
export type {LabelGroupProps} from './LabelGroup'
export {default as Link} from './Link'
export type {LinkProps} from './Link'
export {NavList} from './NavList'
export type {
  NavListProps,
  NavListItemProps,
  NavListSubNavProps,
  NavListGroupProps,
  NavListLeadingVisualProps,
  NavListTrailingVisualProps,
  NavListDividerProps,
} from './NavList'
export {default as Octicon} from './Octicon'
export type {OcticonProps} from './Octicon'
export {default as Overlay} from './Overlay'
export type {OverlayProps} from './Overlay'
export {default as Pagehead} from './Pagehead'
export type {PageheadProps} from './Pagehead'
export {default as Pagination} from './Pagination'
export type {PaginationProps} from './Pagination'
export {default as PointerBox} from './PointerBox'
export type {PointerBoxProps} from './PointerBox'
export {default as Popover} from './Popover'
export type {PopoverProps, PopoverContentProps} from './Popover'
export {default as Portal, registerPortalRoot} from './Portal'
export type {PortalProps} from './Portal'
export {ProgressBar} from './ProgressBar'
export type {ProgressBarProps} from './ProgressBar'
export {default as RadioGroup} from './RadioGroup'
export type {RelativeTimeProps} from './RelativeTime'
export {default as RelativeTime} from './RelativeTime'
export {SegmentedControl} from './SegmentedControl'
export {default as Select} from './Select'
export type {SelectProps} from './Select'
export {SelectPanel} from './SelectPanel'
export type {SelectPanelProps} from './SelectPanel'
export {default as SideNav} from './SideNav'
export type {SideNavProps, SideNavLinkProps} from './SideNav'
export {default as Spinner} from './Spinner'
export type {SpinnerProps} from './Spinner'
export {default as StateLabel} from './StateLabel'
export type {StateLabelProps} from './StateLabel'
export {default as SubNav} from './SubNav'
export type {SubNavProps, SubNavLinkProps, SubNavLinksProps} from './SubNav'
export {default as ToggleSwitch} from './ToggleSwitch'
export {default as TabNav} from './TabNav'
export type {TabNavProps, TabNavLinkProps} from './TabNav'
export {default as TextInput} from './TextInput'
export type {TextInputProps} from './TextInput'
export {default as TextInputWithTokens} from './TextInputWithTokens'
export type {TextInputWithTokensProps} from './TextInputWithTokens'
export {default as Text} from './Text'
export type {TextProps} from './Text'
export {default as Timeline} from './Timeline'
export type {
  TimelineProps,
  TimelineBadgeProps,
  TimelineBodyProps,
  TimelineBreakProps,
  TimelineItemsProps,
} from './Timeline'
export {default as Token, IssueLabelToken, AvatarToken} from './Token'
export type {TokenProps} from './Token'
export {default as Tooltip} from './Tooltip/Tooltip'
export type {TooltipProps} from './Tooltip/Tooltip'
export {default as Truncate} from './Truncate'
export type {TruncateProps} from './Truncate'

export {default as Checkbox} from './Checkbox'
export type {CheckboxProps} from './Checkbox'

export {default as Textarea} from './Textarea'
export type {TextareaProps} from './Textarea'

export {TreeView} from './TreeView'
export type {
  TreeViewProps,
  TreeViewItemProps,
  TreeViewSubTreeProps,
  TreeViewVisualProps,
  TreeViewErrorDialogProps,
} from './TreeView'

export {UnderlineNav} from './UnderlineNav'
export type {UnderlineNavProps, UnderlineNavItemProps} from './UnderlineNav'

export {ActionBar} from './ActionBar'
export type {ActionBarProps} from './ActionBar'

// eslint-disable-next-line no-restricted-imports
export {SSRProvider, useSSRSafeId} from './utils/ssr'
export {default as sx, merge} from './sx'
export type {SxProp} from './sx'
