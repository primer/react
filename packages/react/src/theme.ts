import type {KeyPaths} from './utils/types/KeyPaths'
import {fontStack} from './utils/theme'
import {colorSchemes} from './legacy-theme/ts/color-schemes'

const animation = {
  easeOutCubic: 'cubic-bezier(0.33, 1, 0.68, 1)',
}

const breakpoints = ['544px', '768px', '1012px', '1280px']

const fonts = {
  normal: fontStack([
    '-apple-system',
    'BlinkMacSystemFont',
    'Segoe UI',
    'Noto Sans',
    'Helvetica',
    'Arial',
    'sans-serif',
    'Apple Color Emoji',
    'Segoe UI Emoji',
  ]),
  mono: `var(--fontStack-monospace, ${fontStack([
    'SFMono-Regular',
    'Consolas',
    'Liberation Mono',
    'Menlo',
    'Courier',
    'monospace',
  ])})`,
}

const lineHeights = {
  condensedUltra: 1,
  condensed: 1.25,
  default: 1.5,
}

const fontWeights = {
  light: 300,
  normal: 400,
  semibold: 500,
  bold: 600,
}

const borderWidths = [0, '1px']

const radii = ['0', '3px', '6px', '100px']

const sizes = {
  small: '544px',
  medium: '768px',
  large: '1012px',
  xlarge: '1280px',
}

const fontSizes = ['12px', '14px', '16px', '20px', '24px', '32px', '40px', '48px', '56px']

const space = ['0', '4px', '8px', '16px', '24px', '32px', '40px', '48px', '64px', '80px', '96px', '112px', '128px']

const theme = {
  animation,
  borderWidths,
  breakpoints,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  sizes,
  space,
  colorSchemes,
}

export default theme

// NOTE: for now, ThemeColors and ThemeShadows are handcrafted types. It would be nice if these
// were exports from primitives (or a different shape but derived from those exports).

type ThemeColors = {
  fg: {
    default: string
    muted: string
    subtle: string
    onEmphasis: string
  }
  canvas: {
    default: string
    overlay: string
    inset: string
    subtle: string
  }
  border: {
    default: string
    muted: string
    subtle: string
  }

  // Roles
  neutral: {
    emphasisPlus: string
    emphasis: string
    muted: string
    subtle: string
  }
  accent: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  success: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  attention: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  severe: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  danger: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  open: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  closed: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  done: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  sponsors: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
}

type ThemeShadows = {
  shadow: {
    small: string
    medium: string
    large: string
    extraLarge: string
  }
}

export type ThemeColorPaths = KeyPaths<ThemeColors>
export type ThemeShadowPaths = KeyPaths<ThemeShadows>
