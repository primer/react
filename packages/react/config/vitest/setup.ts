import {afterEach} from 'vitest'
import {cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/vitest'

afterEach(() => {
  cleanup()
})

// @ts-expect-error this is needed for act() from React
// @see https://react.dev/reference/react/act#error-the-current-testing-environment-is-not-configured-to-support-act
globalThis.IS_REACT_ACT_ENVIRONMENT = true
