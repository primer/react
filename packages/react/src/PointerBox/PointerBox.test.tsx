import {describe, expect, it} from 'vitest'
import {render} from '@testing-library/react'
import {PointerBox, ThemeProvider} from '..'
import theme from '../theme'

describe('PointerBox', () => {
  it('applies the border color via "borderColor" prop for backwards compatibility', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <PointerBox borderColor="danger.emphasis" />
      </ThemeProvider>,
    )

    const element = container.firstChild as HTMLElement
    const styles = window.getComputedStyle(element)

    // The borderColor should be applied correctly
    expect(styles.borderColor).toBe('rgb(207, 34, 46)') // danger.emphasis color
  })

  it('applies the border color via sx prop', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <PointerBox sx={{borderColor: 'danger.emphasis'}} />
      </ThemeProvider>,
    )

    const element = container.firstChild as HTMLElement
    const styles = window.getComputedStyle(element)

    // The borderColor should be applied correctly
    expect(styles.borderColor).toBe('rgb(207, 34, 46)') // danger.emphasis color
  })

  it('applies the background color via "bg" prop for backwards compatibility', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <PointerBox bg="danger.emphasis" />
      </ThemeProvider>,
    )

    const element = container.firstChild as HTMLElement
    const styles = window.getComputedStyle(element)

    // The background should include the danger.emphasis color in the gradient
    expect(styles.backgroundImage).toContain('rgb(207, 34, 46)') // danger.emphasis color
  })

  it('applies the background color via sx prop', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <PointerBox sx={{bg: 'danger.emphasis'}} />
      </ThemeProvider>,
    )

    const element = container.firstChild as HTMLElement
    const styles = window.getComputedStyle(element)

    // The background should include the danger.emphasis color in the gradient
    expect(styles.backgroundImage).toContain('rgb(207, 34, 46)') // danger.emphasis color
  })

  it('ensures that background-color set via bg prop and sx output the same for backwards compatibility', () => {
    const mockBg = 'red'
    const {container: containerBg} = render(
      <ThemeProvider theme={theme}>
        <PointerBox bg={mockBg} />
      </ThemeProvider>,
    )
    const {container: containerSx} = render(
      <ThemeProvider theme={theme}>
        <PointerBox sx={{bg: mockBg}} />
      </ThemeProvider>,
    )

    const elementBg = containerBg.firstChild as HTMLElement
    const elementSx = containerSx.firstChild as HTMLElement

    // Both should have the same computed background styles
    const stylesBg = window.getComputedStyle(elementBg)
    const stylesSx = window.getComputedStyle(elementSx)

    expect(stylesBg.backgroundImage).toBe(stylesSx.backgroundImage)
  })
})
