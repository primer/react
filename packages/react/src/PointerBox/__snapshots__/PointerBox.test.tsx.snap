// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointerBox applies the background color via "bg" prop for backwards compatibility 1`] = `
.c0 {
  background-color: var(--borderColor-danger-emphasis,var(--color-danger-emphasis,#cf222e));
  border-width: 1px;
  border-style: solid;
  border-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  border-radius: 6px;
  --custom-bg: var(--borderColor-danger-emphasis,var(--color-danger-emphasis,#cf222e));
  background-image: linear-gradient(var(--custom-bg),var(--custom-bg)),linear-gradient(var(--bgColor-default,var(--color-canvas-default,#ffffff)),var(--bgColor-default,var(--color-canvas-default,#ffffff)));
  position: relative;
}

<div
  className="c0"
>
  <svg
    height={16}
    role="presentation"
    style={
      {
        "left": "50%",
        "marginLeft": -8,
        "pointerEvents": "none",
        "position": "absolute",
        "top": "100%",
      }
    }
    width={16}
  >
    <g
      transform="translate(8,0)"
    >
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--borderColor-danger-emphasis, var(--color-danger-emphasis, #cf222e))"
      />
      <path
        d="M-8,0L0,8L8,0"
        fill="none"
        stroke="var(--borderColor-default, var(--color-border-default, #d0d7de))"
        strokeWidth="1px"
      />
    </g>
  </svg>
</div>
`;

exports[`PointerBox applies the background color via sx prop 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  border-radius: 6px;
  --custom-bg: var(--borderColor-danger-emphasis,var(--color-danger-emphasis,#cf222e));
  background-image: linear-gradient(var(--custom-bg),var(--custom-bg)),linear-gradient(var(--bgColor-default,var(--color-canvas-default,#ffffff)),var(--bgColor-default,var(--color-canvas-default,#ffffff)));
  position: relative;
}

<div
  className="c0"
>
  <svg
    height={16}
    role="presentation"
    style={
      {
        "left": "50%",
        "marginLeft": -8,
        "pointerEvents": "none",
        "position": "absolute",
        "top": "100%",
      }
    }
    width={16}
  >
    <g
      transform="translate(8,0)"
    >
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--borderColor-danger-emphasis, var(--color-danger-emphasis, #cf222e))"
      />
      <path
        d="M-8,0L0,8L8,0"
        fill="none"
        stroke="var(--borderColor-default, var(--color-border-default, #d0d7de))"
        strokeWidth="1px"
      />
    </g>
  </svg>
</div>
`;

exports[`PointerBox applies the border color via "borderColor" prop for backwards compatibility 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-color: var(--borderColor-danger-emphasis,var(--color-danger-emphasis,#cf222e));
  border-radius: 6px;
  position: relative;
}

<div
  className="c0"
>
  <svg
    height={16}
    role="presentation"
    style={
      {
        "left": "50%",
        "marginLeft": -8,
        "pointerEvents": "none",
        "position": "absolute",
        "top": "100%",
      }
    }
    width={16}
  >
    <g
      transform="translate(8,0)"
    >
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0"
        fill="none"
        stroke="var(--borderColor-danger-emphasis, var(--color-danger-emphasis, #cf222e))"
        strokeWidth="1px"
      />
    </g>
  </svg>
</div>
`;

exports[`PointerBox applies the border color via sx prop 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  border-radius: 6px;
  border-color: var(--borderColor-danger-emphasis,var(--color-danger-emphasis,#cf222e));
  position: relative;
}

<div
  className="c0"
>
  <svg
    height={16}
    role="presentation"
    style={
      {
        "left": "50%",
        "marginLeft": -8,
        "pointerEvents": "none",
        "position": "absolute",
        "top": "100%",
      }
    }
    width={16}
  >
    <g
      transform="translate(8,0)"
    >
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0"
        fill="none"
        stroke="var(--borderColor-danger-emphasis, var(--color-danger-emphasis, #cf222e))"
        strokeWidth="1px"
      />
    </g>
  </svg>
</div>
`;

exports[`PointerBox renders a <Caret> in <Box> with relative positioning 1`] = `
.c0 {
  border-width: 1px;
  border-style: solid;
  border-color: var(--borderColor-default,var(--color-border-default,#d0d7de));
  border-radius: 6px;
  position: relative;
}

<div
  className="c0"
>
  <svg
    height={16}
    role="presentation"
    style={
      {
        "left": "50%",
        "marginLeft": -8,
        "pointerEvents": "none",
        "position": "absolute",
        "top": "100%",
      }
    }
    width={16}
  >
    <g
      transform="translate(8,0)"
    >
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0L-8,0Z"
        fill="var(--bgColor-default, var(--color-canvas-default, #ffffff))"
      />
      <path
        d="M-8,0L0,8L8,0"
        fill="none"
        stroke="var(--borderColor-default, var(--color-border-default, #d0d7de))"
        strokeWidth="1px"
      />
    </g>
  </svg>
</div>
`;
